### DISCUSSION TITLE
ELSA embed C# code into an activity in the graphical work flow

### DISCUSSION THREAD

**Discussion by PeterHuang-ST** [elsa-workflows/elsa-core#3208] [Q&A]
Hi

Does ELSA provide an activity that I can embed C# code into? This is useful when we can design the workflow with the graphical interface for most activities and for a few more complex activities, we would like to write it in C#. Thanks

Peter

**Comment by sfmskywalker** [comment]
Hi @PeterHuang-ST 

We don't have that yet, but I've been thinking about adding support for writing activities using C# (as well as other languages such as JavaScript) from the dashboard at some point. 

Right now, you can only create custom activities written in C# by creating classes in your application. These activities then become available to the designer.

**Reply by PeterHuang-ST** [reply]
If we create custom activities C#, would they show up directly in the designer, or do we still need to perform additional work that allow the activities to show up in the graphical designer? If it would just show up in the graphical designer, than its a powerful feature that you should "Market" it more. Any examples where I can look at?

**Reply by sfmskywalker** [reply]
> If we create custom activities C#, would they show up directly in the designer, or do we still need to perform additional work that allow the activities to show up in the graphical designer?

They would show up directly in the designer. Just write the activity and register it with DI. An example can be found here:
- Custom activity: https://github.com/elsa-workflows/elsa-core/blob/master/src/samples/server/Elsa.Samples.Server.Host/Activities/VehicleActivity.cs
- Registering it with DI: https://github.com/elsa-workflows/elsa-core/blob/master/src/samples/server/Elsa.Samples.Server.Host/Startup.cs#L86

It's also documented: https://elsa-workflows.github.io/elsa-core/docs/next/extensibility/extensibility-custom-activities

>  If it would just show up in the graphical designer, than its a powerful feature that you should "Market" it more

I agree, it's quite powerful indeed. I'll keep your suggestion in mind, thanks!

**Reply by truthz03** [reply]
Hi,

I'm currently also searching for an offline usable workflow engine and Elsa is the best I found until yet.
As I found out there is already an activity called "RunJavaScript" but it would also nice to get an "RunC#" activity :)

Thomas
